inc/server/class-filter-blocks-server.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/**
+ * Filter blocks server logic.
+ *
+ * @package ThemeIsle\GutenbergBlocks\Server
+ */
+
+namespace ThemeIsle\GutenbergBlocks\Server;
+
+/**
+ * Class Filter_Blocks_Server
+ */
+class Filter_Blocks_Server {
+
+	/**
+	 * The main instance var.
+	 *
+	 * @var Filter_Blocks_Server
+	 */
+	public static $instance = null;
+
+	/**
+	 * Rest route namespace.
+	 *
+	 * @var Filter_Blocks_Server
+	 */
+	public $namespace = 'themeisle-gutenberg-blocks/';
+
+	/**
+	 * Rest route version.
+	 *
+	 * @var Filter_Blocks_Server
+	 */
+	public $version = 'v1';
+
+	/**
+	 * Initialize the class
+	 */
+	public function init() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register REST API route
+	 */
+	public function register_routes() {
+		$namespace = $this->namespace . $this->version;
+
+		register_rest_route(
+			$namespace,
+			'/filter_blocks',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'filter_blocks' ),
+					'args'                => array(
+						'block' => array(
+							'type'        => 'string',
+							'required'    => true,
+							'description' => __( 'Block namespace.', 'otter-blocks' ),
+						),
+					),
+					'permission_callback' => function () {
+						return current_user_can( 'edit_posts' );
+					},
+				),
+			)
+		);
+	}
+
+	/**
+	 * Filter Blocks
+	 *
+	 * Search WordPress posts for requested block type.
+	 *
+	 * @param mixed $request REST request.
+	 * @since 1.7.0
+	 * @return mixed|\WP_REST_Response
+	 */
+	public function filter_blocks( $request ) {
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return false;
+		}
+
+		$block  = $request->get_param( 'block' );
+		$blocks = array();
+
+		$post_types = get_post_types(
+			array(
+				'public'  => true,
+				'show_ui' => true,
+			)
+		);
+
+		unset( $post_types['attachment'] );
+		array_push( $post_types, 'wp_block' );
+
+		foreach ( $post_types as $key => $post_type ) {
+			$posts = get_posts(
+				array(
+					'posts_per_page' => 100,
+					'post_type'      => $post_type,
+					'meta_query'     => array( // phpcs:ignore WordPress.VIP.SlowDBQuery.slow_db_query_meta_query, WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+						array(
+							'key'   => '_themeisle_gutenberg_block_has_review',
+							'value' => true,
+						),
+					),
+				)
+			);
+
+			foreach ( $posts as $post ) {
+				if ( ! has_blocks( $post->post_content ) ) {
+					$response = new \WP_REST_Response( $blocks, 200 );
+				}
+
+				$post_blocks = parse_blocks( $post->post_content );
+
+				foreach ( $post_blocks as $post_block ) {
+					if ( $block === $post_block['blockName'] ) {
+						if ( 'themeisle-blocks/review' === $post_block['blockName'] ) {
+							if ( isset( $post_block['attrs']['product'] ) && intval( $post_block['attrs']['product'] ) >= 0 && class_exists( 'WooCommerce' ) ) {
```

Comment:
The comparsion table is only for WooCommere users? `class_exists( 'WooCommerce' )` makes it that way.

Reply:
@Soare-Robert-Daniel Yes and no. You found the issue but this part is for the backend only, the same condition exists in the render function which is causing this.